### PR TITLE
Add debug argument to `web.run_app()`

### LINKED ISSUE #3793: Add debug argument to web.run_app()
```
500 Internal Server Error
Traceback:
Traceback (most recent call last):
   ...
NameError: name '...' is not defined
```
need a simple debug=False in web.run_app() like flask for security?

### PR DESCRIPTION
**Pull request by NewUserHa**
## What do these changes do?

Add debug argument to web.run_app() & new_event_loop()

## Are there changes in behavior for the user?

no?

## Related issue number

Resolves #3793

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/web.py [hunk 1]
```diff
@@ -388,7 +389,9 @@ def run_app(app: Union[Application, Awaitable[Application]], *,
             reuse_address: Optional[bool]=None,
             reuse_port: Optional[bool]=None) -> None:
     """Run an app locally"""
-    loop = asyncio.get_event_loop()
+    loop = asyncio.new_event_loop()
```
Comment (webknjaz) [inline aiohttp/web.py]: Why create a new one?

Comment (asvetlov) [reply aiohttp/web.py]: I've asked @NewUserHa to make `run_app` as close to `asyncio.run()` as possible.
Please look at https://github.com/python/cpython/blob/master/Lib/asyncio/runners.py

File: docs/deployment.rst [hunk 2]
```diff
@@ -321,3 +321,11 @@ By default aiohttp uses own defaults::
...
 For more information please read :ref:`Format Specification for Access
 Log <aiohttp-logging-access-log-format-spec>`.
+
+
+Debug Mode
+---------------------
+
+use ``web.run_app(app, debug=True)`` to easily turn on debug messages.
+
+reload need <https://docs.aiohttp.org/en/latest/web_advanced.html#dev-tools>
```
[the remark this answers is not in the document]
Comment (NewUserHa) [reply docs/deployment.rst]: they are both `debug related` document.
putting them together?

[the remark this answers is not in the document]
Comment (asvetlov) [reply docs/deployment.rst]: We *mention* `aiohttp-devtools` in docs, that's it.
There is no need to describe the tool more.
If debug flag behavior affects `aiohttp-devtools` somehow the documentation for `aiohttp-devtools` should reflect this fact.
@samuelcolvin what is your opinion?

[the remark this answers is not in the document]
Comment (samuelcolvin) [reply docs/deployment.rst]: agreed.

We could change the docs and particularly https://demos.aiohttp.org/en/latest/tutorial.html#aiohttp-demos-polls-getting-started to use devtools more, but that's a different issue.

File: docs/deployment.rst [hunk 3]
```diff
@@ -321,3 +321,11 @@ By default aiohttp uses own defaults::
 
 For more information please read :ref:`Format Specification for Access
 Log <aiohttp-logging-access-log-format-spec>`.
+
+
+Debug Mode
+---------------------
+
+Use ``web.run_app(app, debug=True)`` to easily turn on debug messages.
```
Comment (samuelcolvin) [inline docs/deployment.rst]: can we give some more detail here:
* explain that this enables `asyncio.debug`
* explain what `debug messages` means. I think it just refers to tracebacks, but is there any other changes to messaging/logging?

I think we also need explain that if you're using lower level components like `AppRunner` that debug is infered from `asyncio.debug`?

File: aiohttp/web.py (hunk 1 above)
Comment (webknjaz) [reply aiohttp/web.py]: Alright, so perhaps a comment in the code explaining the motivation would be helpful

File: docs/deployment.rst (hunk 3 above)
Comment (asvetlov) [reply docs/deployment.rst]: Agree.
It enables asyncio debug and switches on access log printing if the `aiohttp.access` logger was not configured earlier explicitly.

@samuelcolvin seems like @NewUserHa is not a native speaker, writing English texts is a complex task for him.
Would you provide a comment with an example of the proposed text to copy-paste it into the commit?

Comment (samuelcolvin) [reply docs/deployment.rst]: Ok, but I think this is the wrong place to document this behaviour, personally I think we should:
* add a comment to [quickstart](https://docs.aiohttp.org/en/stable/web_quickstart.html) saying "you might want to use `debug=True` to see tracebacks, there's also aiohttp-devtools if you're looking for code reloading functionality"
* give a fuller description of how debug works in the [`run_app` function documentation](https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.run_app)

If you agree on that I'm happy to write the text but it sounds like another PR would be the easiest way to do it?

Comment (asvetlov) [reply docs/deployment.rst]: Agree, a separate PR would be easier for you to make it done.

Let's drop this section here.
Note: `debug` flag should be enumerated in parameters list for `run_app` in `web_reference` in by PR. Later @samuelcolvin provide a narrative text for the flag usage.

File: aiohttp/web.py [hunk 4]
```diff
@@ -388,7 +389,10 @@ def run_app(app: Union[Application, Awaitable[Application]], *,
             reuse_address: Optional[bool]=None,
             reuse_port: Optional[bool]=None) -> None:
     """Run an app locally"""
-    loop = asyncio.get_event_loop()
+    # behavior like asyncio.run() as a main entry point for asyncio programs
```
Comment (asvetlov) [inline aiohttp/web.py]:
```suggestion
    # The function design follows asyncio.run() behavior
```

File: docs/web_reference.rst [hunk 5]
```diff
@@ -2725,6 +2725,8 @@ Utilities
    :param app: :class:`Application` instance to run or a *coroutine*
                that returns an application.
 
+   :param bool debug: same as `loop.set_debug(True)`
```
Comment (asvetlov) [inline docs/web_reference.rst]:
```suggestion
   :param bool debug: enable :ref:`asyncio debug mode <asyncio-debug-mode>` if ``True``.
```

**Comment by asvetlov** [comment]
The code looks good.
Please take a look at https://travis-ci.com/aio-libs/aiohttp/jobs/203039958 
`make flake` command fails :(

**Comment by NewUserHa** [comment]
thanks
it passes now

**Comment by asvetlov** [comment]
Good!
The next and last step is fixing tests.
`make test` hangs now :(

**Comment by NewUserHa** [comment]
`tests/test_run_app.py::test_run_app_http[pyloop] ` I'm asked to add `asyncio.new_event_loop()` :(

**Comment by NewUserHa** [comment]
but this seems about argument position.
what about putting debug flag at the last, again?

but that's keyword-only argument in the test hanged. how did last build success?

**Comment by asvetlov** [comment]
The code from PR is correct, tests should be updated to work with it.
All tests that checks some network and thus relies on `patched_loop` can be easily switched from testing `web.run_app()` to internal `web._run_app()`

**Comment by NewUserHa** [comment]
simply replacing web.run_app() with web._run_app() doesn't work

**Comment by asvetlov** [comment]
@NewUserHa if you have no time/energy to finish the PR -- that's sad but ok.
I have no estimation when I'll have a time window to work on the issue.
It is not in my current very long to-do list yet, sorry.
Maybe somebody else can pick it up and make it done.

Mechanical `run_app()` -> `_run_app()` in tests is not a solution.
You should iterate over all failed tests to figure out how to fix a failure. There is no generic and easy way to make everybody happy.

**Comment by NewUserHa** [comment]
no.
since I'm not familiar with asyncio test, I'm waiting if you have ideas for this. because you said `can be easily switched from testing web.run_app() to internal web._run_app()`

need I go to find why by myself?

File: aiohttp/web.py [hunk 6]
```diff
@@ -388,7 +389,9 @@ def run_app(app: Union[Application, Awaitable[Application]], *,
             reuse_address: Optional[bool]=None,
             reuse_port: Optional[bool]=None) -> None:
     """Run an app locally"""
-    loop = asyncio.get_event_loop()
+    # The function design follows asyncio.run() behavior
```
Comment (webknjaz) [inline aiohttp/web.py]: ....that is why....?

Comment (NewUserHa) [reply aiohttp/web.py]: asvetlov asked it

**Comment by NewUserHa** [comment]
I can't get the change of `new_event_loop` pass the test, so get back to the original issue that only change the debug argument first.

**Pull request merged**